trace/trace.go

```diff
@@ -472,3 +580,39 @@ func (s *Span) createChildSpan(ctx context.Context, async bool) (context.Context
 	ctx = PutSpanInContext(ctx, newSpan)
 	return ctx, newSpan
 }
+
+type randomIdentifier struct {
+	randomBytes   func([]byte) (int, error)
+	incrementLock sync.Mutex
+	nonRandomId   uint64
+}
+
+// newRandomIdentifier returns a new randomIdentifier with the default pseudo-random
+// generator.
+func newRandomIdentifier() *randomIdentifier {
+	return &randomIdentifier{
+		randomBytes: rand.Read,
+	}
+}
+
+// build generates random bytes and encodes them as a hex string. The
+// length of the id depends on the length of the byte slice provided. If
+// rand.Read fails, the result will be a best attempt based on a counter.
+func (r *randomIdentifier) build(id []byte) string {
+	_, err := r.randomBytes(id)
+	if err != nil {
+		nonRandomId := make([]byte, len(id))
```

Comment:
why can't we use continue to use `id` here for this (instead of creating a new `[]byte`)?